### ISSUE TITLE
Exporting ONNX Model with Final Class Probabilities Instead of Embeddings

### ISSUE THREAD

**Issue by didi1233** [wenet-e2e/wespeaker#340]
Dear WeSpeaker Team,

I am trying to use WeSpeaker for a classification task. I have trained a three-class model using ResNet and a Linear classifier. However, I would like the exported ONNX model to output the final class probabilities instead of the embeddings. I attempted to use the following code, but the inference results are very incorrect. Could you please help me identify the problem? Thank you.

```
from __future__ import print_function

import argparse
import numpy as np
import torch
import torch.nn as nn
import yaml

from speaker_model import get_speaker_model
from checkpoint import load_checkpoint
from resnet import ResNet18

class Linear(nn.Module):
    """
    The linear transform for simple softmax loss
    """

    def __init__(self, emb_dim=256, class_num=3):
        super(Linear, self).__init__()

        self.trans = nn.Sequential(nn.BatchNorm1d(emb_dim),
                                   nn.ReLU(inplace=True),
                                   nn.Linear(emb_dim, class_num))

    def forward(self, input, label=None):
        out = self.trans(input)
        return out

class CombinedModel(nn.Module):
    def __init__(self, resnet_model, classifier):
        super(CombinedModel, self).__init__()
        self.resnet_model = resnet_model
        self.classifier = classifier

    def forward(self, x):
        _, embeddings = self.resnet_model(x)
        output = self.classifier(embeddings)
        return output

def get_args():
    parser = argparse.ArgumentParser(description='export your script model')
    parser.add_argument('--config', required=True, help='config file')
    parser.add_argument('--checkpoint', required=True, help='checkpoint model')
    parser.add_argument('--output_model', required=True, help='output file')
    parser.add_argument('--mean_vec',
                        required=False,
                        default=None,
                        help='mean vector')
    args = parser.parse_args()
    return args

def main():
    args = get_args()

    with open(args.config, 'r') as fin:
        configs = yaml.load(fin, Loader=yaml.FullLoader)
    resnet_model = ResNet18(feat_dim=80, embed_dim=256, two_emb_layer=False)
    classifier = Linear(emb_dim=256, class_num=3)
    model = CombinedModel(resnet_model, classifier)
    load_checkpoint(model, args.checkpoint)
    model.eval()

    feat_dim = configs['model_args'].get('feat_dim', 80)
    if 'feature_args' in configs:  # deprecated IO
        num_frms = configs['feature_args'].get('num_frms', 200)
    else:  # UIO
        num_frms = configs['dataset_args'].get('num_frms', 200)

    dummy_input = torch.ones(1, num_frms, feat_dim)
    torch.onnx.export(model,
                      dummy_input,
                      args.output_model,
                      do_constant_folding=True,
                      verbose=False,
                      opset_version=14,
                      input_names=['feats'],
                      output_names=['embs'],
                      dynamic_axes={
                          'feats': {
                              0: 'B',
                              1: 'T'
                          },
                          'embs': {
                              0: 'B'
                          }
                      })

if __name__ == '__main__':
    main()
```

**Comment by JiJiJiang** [comment]
Thank you for your question.
We only save the model part before the speaker embedding layer in WeSpeaker, while the classifier (from embedding to speaker label) is not saved during training.
For your case, you should modify the codes in save_checkpoint and save the classifier, otherwise the classifier would be with random parameters and thus the final results would be random too.

**Comment by JiJiJiang** [comment]
@didi1233 Have you solved this problem? My answer is not correct above. We did save both the whole speaker model including the classifier you need, but during extracting, we did not load the classifier part.

For your case, if you train the model with the naive softmax loss, I think it should work well using our training pipeline.

**Comment by JiJiJiang** [comment]
I see. Thanks for your answer.
